### PR TITLE
chore: update to setup.cfg

### PR DESCRIPTION
**Pull request by henryiii**
Converting to setup.cfg. Automatic + minor touchup manually.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,14 +1,72 @@
...
+    decay
+    chain
+    representation
+
+[options]
+packages = find:
+install_requires =
+    attrs>=19.2
+    graphviz
+    lark-parser>=0.8.0,<0.8.6
```
Comment (henryiii) [inline setup.cfg:42]: We need to fix this to work with newer versions...

File: setup.cfg [hunk 2]
```diff
@@ -1,14 +1,72 @@
...
+    pandas>=0.22
+    particle==0.14.*
+    plumbum>=1.6.9
+    six>=1.11
+    cachetools;python_version<"3.3"
+    enum34>=1.1;python_version<"3.4"
+    importlib_resources>=1.0;python_version<"3.7"
+    pathlib2>=2.3;python_version<"3.5"
+python_requires = >=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*
+include_package_data = True
```
Comment (henryiii) [inline setup.cfg:53]: This seems simpler than what was there before.

File: setup.cfg [hunk 3]
```diff
@@ -1,14 +1,72 @@
...
+[options.extras_require]
+test =
+    pytest>=4.6
+
 [bdist_wheel]
 universal = 1
 
-[aliases]
-test=pytest
-
```
Comment (henryiii) [inline setup.cfg:6]: Deprecated, users should not be calling `python setup.py <anything>`

File: setup.cfg (hunk 1 above)
Comment (eduardo-rodrigues) [reply setup.cfg:42]: Indeed. I look at that a while back and concluded at the time that Lark had changed behaviour significantly, in a way that almost looked like a bug. I can't remember the details now but then decided not to dwell on it immediate. I agree that we need to move to recent versions, also to potentially make things work for latest Python versions. I will create an issue ...

File: setup.cfg (hunk 3 above)
Comment (eduardo-rodrigues) [reply setup.cfg:6]: Fair.

File: setup.cfg (hunk 2 above)
Comment (eduardo-rodrigues) [reply setup.cfg:53]: Indeed. No need for a temporary variable. Thanks.

**Comment by eduardo-rodrigues** [comment]
The issue with Python 3.5 is solvable for sure but I'm perfectly happy to drop support for it in the next minor release, so ...

**Comment by henryiii** [comment]
Weird, why is lark missing on 3.5?

**Comment by eduardo-rodrigues** [comment]
It isn't as I can see from the CI from my PR. So maybe a weird glitch? In any case it seems we can get rid of 3.5 anyway at this stage :-).

**Comment by eduardo-rodrigues** [comment]
The CI for Python 3.5 in the step "Install package" says
```
Installing collected packages: UNKNOWN
  Running setup.py develop for UNKNOWN
Successfully installed UNKNOWN
Finishing: Install package
```
whereas for the other versions is starts collecting and installing the required packages.

File: setup.cfg [hunk 4]
```diff
@@ -1,14 +1,72 @@
+[metadata]
+name = DecayLanguage
+description = A language to describe particle decays, and tools to work with them.
```
Comment (eduardo-rodrigues) [inline setup.cfg:3]:
```suggestion
description = A language to describe, manipulate and convert particle decays.
```

**Comment by eduardo-rodrigues** [comment]
OK, GA also fails on Python 3.5. So it's got to do with our setup. The only thing I see removed is `tests_require=extras["test"],`.

**Comment by henryiii** [comment]
Ahh, pyproject.toml is missing. I think that’s the problem.

**Comment by eduardo-rodrigues** [comment]
> Ahh, pyproject.toml is missing. I think that’s the problem.

Dunno. Will you take care of adding one?

**Review by eduardo-rodrigues — APPROVED** [review]
LGTM

(Noted that there is more on Scikit-HEP guidelines but this one PR is scoped and ready).

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+[build-system]
+requires = [
+    "setuptools>=42",
+    "wheel",
+    "setuptools_scm[toml]>=3.4"
```
Comment (henryiii) [inline pyproject.toml:5]: We don't use `setuptools_scm` (yet).
```suggestion
```

**Comment by henryiii** [comment]
Yes, one step at a time. If you want to add one of the other parts (setuptools_scm, more checks, etc), feel free. I'll try to ping you if I start working on one.

**Comment by eduardo-rodrigues** [comment]
> Yes, one step at a time. If you want to add one of the other parts (setuptools_scm, more checks, etc), feel free. I'll try to ping you if I start working on one.

OK, sounds good. I actually found a bug in decay chain representations so will probably focus on that first, tough; for next week.

**Pull request merged**